scw/lister.go

```diff
@@ -0,0 +1,54 @@
+package scw
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+)
+
+type lister interface {
+	GetTotalCount() int
+	Append(interface{}) (int, error)
+}
+
+// doListAll collect all pages of a List request and aggregate all results on a single response.
+func (c *Client) doListAll(req *ScalewayRequest, res interface{}) (err error) {
+	switch response := res.(type) {
+	case lister:
+		n := 0
+		page := 1 // start at page 1
+		totalCount := -1
+		for i := 0; i < totalCount; i += n {
+			// set current page
+			req.Query.Set("page", strconv.Itoa(page))
```

Comment:
We should avoid mutate the request (make a copy ?)

Reply:
I can copy the query parameters at the beginning and set it back at the end, does it work for you?